### PR TITLE
New PR for subscribe overlay

### PR DESCRIPTION
**Pull request by VirtualRabb**
Add subscribe overlay

# Description

I've created a subscribe overlay for the end of the videos.
It adds a overlay with a subscribe and hit the bell animation to the end of the video which can be seen in the video below

https://youtube.com/shorts/6f6iomWNpaw?feature=share

New setting in config for enabling and disabling the overlay
If it's disabled from the start it will not download or create the folders required to use the overlay

-- Choose between multiple overlays
I've added a new setting within the config file, where you can choose between 
youtube, tiktok and instagram overlays!

-- Moved the add overlay method into the proper class
I've the thing that composites the overlay onto the video from the final_video.py to video.py

It uses google drive to store the file for download

# Checklist:

- [X] I am pushing changes to the **develop** branch
- [X] I am using the recommended development environment
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have formatted and linted my code using python-black and pylint
- [X] I have cleaned up unnecessary files
- [X] My changes generate no new warnings
- [X] My changes follow the existing code-style
- [X] My changes are relevant to the project

# Any other information (e.g how to test the changes)

Enable the sub_overlay in config or on startup.

### CODE REVIEW

File: video_creation/subOverlay.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            fileName = "instagram"
+
+        if Path(f"assets/subOverlay/{fileName}.mov").is_file():
+            return
+
+        #Generate path and folders
+        if not path.isdir(f'assets/subOverlay'):
+            makedirs(f'assets/subOverlay')
+
+        print_step("We need to download the subscribe overlay, this only needs to be done once!")
```
[the remark this answers is not in the document]
Comment (VirtualRabb) [reply video_creation/subOverlay.py:32]: > The files are in this PR. So you can remove the download part

Ah ye true, I'll get rid of that right away :d

[the remark this answers is not in the document]
Comment (StopmotionSimonYT) [reply video_creation/subOverlay.py]: Awesome!

**Review by StopmotionSimonYT — APPROVED** [review]
Good pull request, glad you got the merge conflicts fixed

**Review by callumio — APPROVED** [review]
Looks good, I'll merge into develop - need to see if there is a performance difference

**Comment by callumio** [comment]
Also please resolve conflicfs

**Comment by VirtualRabb** [comment]
> Also please resolve conflicfs

ffs Git no like me :;d

**Comment by VirtualRabb** [comment]
I cannot find the conflicts when I do git status it says this

https://gyazo.com/6d45e6fe173740fdd5860b515ad40d66

And there's no conflict markers anywhere to be found.-.
Is there something I'm doing wrong?

**Comment by VirtualRabb** [comment]
> Also please resolve conflicfs

I can't find where that should be at, I've looked everywhere now .-.

**Comment by StopmotionSimonYT** [comment]
> > Also please resolve conflicfs
> 
> I can't find where that should be at, I've looked everywhere now .-.

I can try to copy your pull request and give you credit for it?

**Comment by VirtualRabb** [comment]
> > > Also please resolve conflicfs
> > 
> > 
> > I can't find where that should be at, I've looked everywhere now .-.
> 
> I can try to copy your pull request and give you credit for it?

Go ahead :)

**Comment by StopmotionSimonYT** [comment]
Okay, I have a plan. Can you revert all the changes and exclude the video files? So that all the code is unchanged. This is because my git doesn't work for some reason, so I can't upload the video files.  If that is done, I will make a pull request with the correct code. Then this pull request and my pull request can be merged, and the problem is solved.

**Comment by VirtualRabb** [comment]
> Okay, I have a plan. Can you revert all the changes and exclude the video files? So that all the code is unchanged. This is because my git doesn't work for some reason, so I can't upload the video files. If that is done, I will make a pull request with the correct code. Then this pull request and my pull request can be merged, and the problem is solved.

I had the same issue it's because in the .gitignore it ignores assets/ so i removed that got the videos and folders uploaded and readded it :d
But yeah I can revert all the code real quick :d

**Comment by VirtualRabb** [comment]
> > Okay, I have a plan. Can you revert all the changes and exclude the video files? So that all the code is unchanged. This is because my git doesn't work for some reason, so I can't upload the video files. If that is done, I will make a pull request with the correct code. Then this pull request and my pull request can be merged, and the problem is solved.
> 
> I had the same issue it's because in the .gitignore it ignores assets/ so i removed that got the videos and folders uploaded and readded it :d But yeah I can revert all the code real quick :d

I've reverted all the code :d

**Comment by StopmotionSimonYT** [comment]
I managed to add the videos to my PR!

**Comment by JasonLovesDoggo** [comment]
see #1196

**Pull request closed without merging**